pkg/3scale/amp/operator/upgrade.go

```diff
@@ -480,10 +480,16 @@ func (u *UpgradeApiManager) upgradeSystemS3EnvVars() (reconcile.Result, error) {
 }
 
 func (u *UpgradeApiManager) upgradeAwsSecret() (reconcile.Result, error) {
+	if u.Cr.Spec.System.FileStorageSpec.DeprecatedS3 == nil {
+		// We check whether S3 section has been migrated. If it has we assume
+		// everything went ok
+		return reconcile.Result{}, nil
```

Comment:
See this

Reply:
check comment above. I think this check has no sense as has been checked before to reach `upgradeAwsSecret()`

Reply:
I think this might still be needed.
When the methods make changes they return that want to requeue and the exeuction stops at that point, requeuing the event and reevaluating the logic. So, the first time upgradeAwsSecret is executed DeprecatedS3 is not nil but it will be reexecuted again when it is nil, when the requeueing due to upgradeApimanagerCRS3Attrs is produced. It should be "idempotent"

Reply:
I'll remove this check as we have prevented the upgradeSystemS3 to reenter after updating the CR sections at the end